include/picongpu/fields/MaxwellSolver/CKC/CKC.hpp

```diff
@@ -0,0 +1,90 @@
+/* Copyright 2013-2023 Axel Huebl, Heiko Burau, Rene Widera, Remi Lehe,
+ *                     Sergei Bastrakov, Lennert Sprenger
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "picongpu/simulation_defines.hpp"
+
+#include "picongpu/fields/MaxwellSolver/CFLChecker.hpp"
+#include "picongpu/fields/MaxwellSolver/CKC/CKC.def"
+#include "picongpu/fields/MaxwellSolver/CKC/Derivative.hpp"
+#include "picongpu/fields/MaxwellSolver/DispersionRelation.hpp"
+#include "picongpu/fields/MaxwellSolver/FDTD/FDTD.hpp"
+
+#include <pmacc/algorithms/math/defines/pi.hpp>
+#include <pmacc/traits/GetStringProperties.hpp>
+
+
+namespace picongpu
+{
+    namespace fields
+    {
+        namespace maxwellSolver
+        {
+            /** Specialization of the CFL condition checker for CKC solver
+             *
+             * @tparam T_Defer technical parameter to defer evaluation
+             */
+            template<typename T_Defer>
+            struct CFLChecker<CKC, T_Defer>
+            {
+                /** Check the CFL condition according to the paper, doesn't compile when failed
+                 *
+                 * @return value of 'X' to fulfill the condition 'c * dt <= X'
```

Comment:
It was hard to understand for me what `X` is. Only after reading the code, I understand what the return value is. I would rephrase it into something like this
```suggestion
                 * @return shortest edge length of the cell (necessarily fulfilling `>= c * dt` because CFL is fulfilled if this compiled at all)
```

Reply:
It was called `X` since it was phrased like this in other solvers (e.g.[Yee Solver](https://github.com/ComputationalRadiationPhysics/picongpu/blob/5456d097efc3c6e60bb395e11ff452a3b280963e/include/picongpu/fields/MaxwellSolver/Yee/Yee.hpp#L48)) because i oriented myself on that. In general `X` is just a value dependent on the solver and the grid. It is just in this special implementation that it is the shortest edge length.
So this could also be a new general discussion on how to phrase it in the corresponding CFL conditions for the different solvers.

Reply:
Okay, I give you that. I still think that this is badly phrased (also in the Yee case) but you're off the hook. I've created #4815 to track this. Please come back to this and adjust if my suggestion gets merged before your PR. Otherwise, you don't have to act on this.